run_unittests.py

```diff
@@ -5665,6 +5665,20 @@ def test_pkgconfig_link_order(self):
         deps = stdo.split()
         self.assertTrue(deps.index(b'-lsomething') < deps.index(b'-ldependency'))
 
+    @skipIfNoPkgconfig
+    def test_pkgconfig_cross_linkdir(self):
+        '''
+        Test that libraries are listed before their dependencies.
+        '''
+        testdir = os.path.join(self.unit_test_dir, '73 cross pkgconfig libdir')
+        self.init(testdir, extra_args=['-Dgenerate=true'])
```

Comment:
Seems overkill to call meson just to generate a fil, see for example how it's done in `test_native_dep_pkgconfig()`, if you set `self.meson_cross_file` it will automatically be passed as argument when you call `self.init()`

Reply:
I dont think its overkill seen that I need to generate the crossfile with pkg_config_libdir

Reply:
I mean you can generate it with python code in run_unittest.py instead of generating it in meson.build code. See test_native_dep_pkgconfig().

Reply:
I got inspiration from  `test_identity_cross`, is it bad ?

Reply:
oh I see... I'm sorry to have misleading you, but now that I saw test_native_dep_pkgconfig(), I think it does it better. That test exactly the same thing as you want, except it uses a pkg-config wrapper to set PKG_CONFIG_LIBDIR. I think it would make sense to add a variant of that test that uses pkg-config without the wrapper but set pkg_config_libdir in the cross file. That way you can even keep using `46 native dep pkgconfig var` instead of writing a new meson.build file, etc.

Reply:
ok I see.

Reply:
I adapt the test to use now the new property and removed the 73 test I introduced before. Hope its fitting your expectations